ra/runtime/src/main/java/io/quarkus/artemis/jms/ra/ArtemisResourceAdapterFactory.java

```diff
@@ -43,8 +43,19 @@ public String getProductVersion() {
     @Override
     public ActiveMQResourceAdapter createResourceAdapter(String id, Map<String, String> config) {
         ActiveMQResourceAdapter adapter = new ActiveMQResourceAdapter();
-        adapter.setConnectorClassName(NettyConnectorFactory.class.getName());
-        adapter.setConnectionParameters(config.get("connection-parameters"));
+        String connectionParameters = config.get("connection-parameters");
+        int hosts = Math.max(1, count(connectionParameters, "host="));
+        // Repeat the NettyConnectorFactory class name for each host
+        String connectorClassName = NettyConnectorFactory.class.getName();
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < hosts; i++) {
+            if (i > 0) {
+                sb.append(",");
+            }
+            sb.append(connectorClassName);
+        }
+        adapter.setConnectorClassName(sb.toString());
```

Comment:
WDYT?
```suggestion
        adapter.setConnectorClassName(IntStream.range(1, hosts)
            .mapToObj(unused -> connectorClassName)
            .collect(Collectors.joining(",")));
```

Reply:
Not sure introducing lambdas here for something simple is a good thing, maybe for readability, but I have no objections if the team agrees on it

Reply:
What exactly is the issue with the lambda?

Reply:
I don't mind either. Since this is only run once we can prefer readability over performance. You can do `sb.append(',')` and skip the `if` and do a `subString(0, sb.length() - 1)` at the end.

Reply:
> What exactly is the issue with the lambda?

It's a new class created in disk which gets instantiated and consumes memory
 
> Anyway, please do not treat this as a blocker. I tend to use streams for readability, but I am absolutely fine with keeping the loop.

+1

Reply:
Sorry, When we compile the lambda expression, an equivalent private static or non-static method is added to the bytecode, the information above is incorrect

Reply:
Well, IIRC, this ResourceAdaptorFactory could be initilized at the build time. So I don't see much performance issues here. +1 for the lambda.